### PR TITLE
fix: verified matches are not visible for participants when matchesFi…

### PR DESCRIPTION
**Pull request by yuvanadarsh**
# Pull Request

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [X] Other (please describe): No change made to code, accidentally reformatted while editing but opened a PR to show that the bug was not relevant.

## Related Issues (put task name here from Notion)
Bug: Dashboard Message Mismatch

### Code Review Preparation
- [X] I have performed a self-review and reviewed the changed files
- [X] My code follows the project's style guidelines and passes linting
- [X] I have added comments in hard-to-understand areas and updated documentation

### Testing
- [X] I have added relevant tests and they pass locally
- [X] I have tested the changes manually, including edge cases

### Code Quality & Security
- [X] I have formatted my code and removed debugging/unused code
- [X] No TypeScript errors or linting issues
- [X] No sensitive information committed (API keys, passwords, etc.)
- [X] User inputs are validated and sanitized

## Additional Notes
To reiterate, No code was modified, everything seems to be correct and as asked of.

Bug report:
For the participant profiles, the dashboard page states that admins are still in the process of matching, but the participant was already matched. 

Ensure matches are not visible for participants when matchesFinal = false, and are visible (but still only shows waiting page) when matchesFinal = true.

Code snippet in question: 
```typescript
if (!matches_final && !started) {
      statusTitle = "We’re still finalizing matches";
      statusBody =
        "Thank you for filling out the registration form! We’re still working on finalizing everyone’s matches. You’ll receive an update as soon as your match is ready.";
    } else if (!matches_final && started) {
      statusTitle = "We’re still finalizing matches";
      statusBody =
        "Thank you for filling out the registration form! We’re still working on finalizing everyone’s matches. You’ll receive an update as soon as your match is ready.";
    } else if (matches_final && !started) {
      statusTitle = "Your match has been set!";
      statusBody =
        "Your match has been finalized. You’ll get full access to the program once it officially starts. Keep an eye on your email for the start date and next steps.";
    } else if (matches_final && started) {
      statusTitle = "Redirecting you to your dashboard…";
      statusBody = "";
    }
```

## Summary by CodeRabbit

* **New Features**
  * Enhanced the Waiting page’s match details view: shows a dedicated “Loading match info…” state while data is pending.
  * When a partner is available, displays their profile information and contact number in the match section.
  * When no partner is available, shows a single-line admin contact message with a prefilled email link.
* **Bug Fixes**
  * Improved consistency and readability of status messaging across waitlist, program, and unavailable states.

### CODE REVIEW

File: src/pages/Waiting/Waiting.tsx [hunk 1]
```diff
@@ -35,41 +20,23 @@ function buildGreetingName(
...
+  const { user, participant, loading, participantLoading, programState, programStateLoading, isWaitlisted, waitlistLoading } = useAuth();
 
   const navigate = useNavigate();
 
   const [match, setMatch] = useState<(Match & { id: string }) | null>(null);
   const [partner, setPartner] = useState<PartnerInfo | null>(null);
   const [matchLoading, setMatchLoading] = useState(false);
 
-
   const isStudent = participant?.user_type?.toLowerCase() === "student";
```
[the remark this answers is not in the document]
Comment (evtran4) [reply src/pages/Waiting/Waiting.tsx]: Yup. Please remove any dead code.

**Review by evtran4 — COMMENTED** [review]
Looks good! Just remove the unused code pls

**Review by evtran4 — APPROVED** [review]
